docs/rfcs/0001-ssl-transportation.md

```diff
@@ -0,0 +1,79 @@
+# Summary
+
+Introduce the SSL/TLS protocol for secure data transport.
+
+# Motivation
+
+Crypto the data transportation.
+
+# Usage explanation
+
+Define some global flags:
+
+```c++
+DEFINE_string(cert_path, "", "Path to cert pem.");
+DEFINE_string(key_path, "", "Path to cert key.");
+DEFINE_string(ca_path, "", "Path to trusted CA file.");
+DEFINE_bool(enable_ssl, false, "Whether enable ssl.");
+DEFINE_bool(enable_graph_ssl, false, "Whether enable ssl in graph server only.");
+DEFINE_bool(enable_meta_ssl, false, "Whether enable ssl in meta server only.");
+```
+
+Two certificate mode:
+
+1. Self Signed: only provide `cert_path` and `key_path`.
+2. CA Signed: provide `cert_path`, `key_path` and `ca_path`.
+
+And provide three policies to control crypto components;
+
+1. Encrypt the whole data transportation in the cluster when the flag `enable_ssl` is set to `true`. In this mode, we enable SSL/TLS transportation for following components of daemon:
+
+```plain
+graphd: meta client + storage client + graph server
+
+metad: admin storage client + metad server + raft client + raft server
+
+storaged: meta client + internal storage client + internal storage server + graph storage server + admin storage server + raft client + raft server
+```
+
+2. Encrypt the graph server when the flag `enable_graph_ssl` is set to `true`. In this mode, we enable SSL/TLS transportation for follow components of daemon:
+
+```plain
+graphd: graph server
+```
+
+This policy is optimized for the case that the cluster setup in one room and communicate over the internal network, and only graph server expose the interface to outside.
+
+3. Encrypt the meta server `enable_meta_ssl` is set to `true`. In this mode, we enable SSL/TLS transportation for follow components of daemon:
+
+```plain
+graphd: meta client
+
+metad: metad server
+
+storaged: meta client
+```
+
+We provide this control policy for meta will transport some secret information.
```

Comment:
might be more clear if change "will" to "to"?

Reply:
Good catch.